### PR TITLE
P3 — Planner Test Coverage (Manifests, Merge, Compiler)

### LINKED ISSUE #1540: P3: Planner test coverage (manifests, merge, compiler)
## Goal

Add error-branch and edge-case test coverage for the planner subsystem: manifest builders (`build_pre_elab_snapshot`, `build_phase_assignment_manifest`, `build_phase_wp_manifest`, `finalize_wp_manifest`), merge utilities (`merge_tier_dir`, `replace_item`), `create_run_dir`, and `compile_plan`.

Consolidates: P3-A3, P3-A4

## Context

- Phase: P3 — Test Coverage Expansion
- Depends on: None
- Depended on by: None

---

## WP1: build_pre_elab_snapshot tests (from A3)

Add 4 tests to `test_manifests.py`:
- Happy path: valid manifest with 2 items, phases sorted by ordering
- Corrupt JSON: `json.JSONDecodeError` raised
- Missing file: `FileNotFoundError` raised
- Empty items: `phases == []`

## WP2: create_run_dir empty string (from A3)

Add 1 test to `test_planner.py`:
- `create_run_dir('')` → `ValueError` matching `'non-empty'`

## WP3: merge_tier_dir tests (from A3)

Add 2 tests to `test_merge.py`:
- Empty dir → `ValueError` matching `'No *_result.json files found'`
- Single file → `item_count='1'` and correct output

## WP4: replace_item error branches (from A3)

Add 2 tests to `test_merge.py`:
- Nonexistent replacement file → `ValueError` matching `'Replacement file not found'`
- Corrupt JSON → `ValueError` matching `'Invalid JSON'`

## WP5: Manifest builder error branches (from A3)

Add 9 tests to `test_manifests.py` covering three callables:
- `build_phase_assignment_manifest`: corrupt JSON, missing required keys, empty string dir
- `build_phase_wp_manifest`: corrupt JSON, empty string, nonexistent dir
- `finalize_wp_manifest`: corrupt JSON, empty string, nonexistent dir

## WP6: compile_plan edge cases (from A4)

Add `TestCompilePlanEdgeCases` to `test_compiler.py`:
- `verdict='fail'` in `validation.json` → logs warning, returns dict (no exception)
- WP references absent phase P99 → `RuntimeError` matching `'references phase'`
- WP references absent assignment P1-A99 → `RuntimeError` matching `'references assignment'`
- Malformed WP id `'BADID'` → `ValueError` from `_parse_wp_id`

---

## Acceptance Criteria

- [ ] ~18 test methods across 4 test files
- [ ] All error branches covered: `json.JSONDecodeError`, `ValueError`, `FileNotFoundError`, `RuntimeError`
- [ ] No source files modified — tests only
- [ ] All tests xdist-safe under `pytest -n 4`
- [ ] `task test-all` passes

### PR DESCRIPTION
**Pull request by Trecek**
## Summary

Add ~22 test methods across 4 existing test files covering error branches and edge cases in the planner subsystem: `build_plan_snapshot`, `create_run_dir`, `merge_tier_dir`, `replace_item`, manifest builders (`build_phase_assignment_manifest`, `build_phase_wp_manifest`, `finalize_wp_manifest`), and `compile_plan`. No source files are modified — tests only.

**Issue correction:** The issue references `build_pre_elab_snapshot` (WP1) but this function does not exist. The correct function is `build_plan_snapshot` in `src/autoskillit/planner/merge.py`. Existing tests for this function already live in `tests/planner/test_merge.py`, so WP1 tests are placed there (not in `test_manifests.py` as the issue suggests).

**Behavioral correction (WP1):** The issue states corrupt JSON raises `json.JSONDecodeError` and missing file raises `FileNotFoundError`. In reality, `build_plan_snapshot` **swallows** both errors via a `try/except (ValueError, json.JSONDecodeError)` block and logs a warning. Tests must verify the actual skip-and-warn behavior, not exception propagation.

Closes #1540

## Implementation Plan

Plan file: `/home/talon/projects/autoskillit-runs/impl-20260430-081121-173068/.autoskillit/temp/make-plan/p3_planner_test_coverage_plan_2026-04-30_082100.md`

🤖 Generated with [Claude Code](https://claude.com/claude-code) via AutoSkillit

## Token Usage Summary

| Step | uncached | output | cache_read | cache_write | count | time |
|------|----------|--------|------------|-------------|-------|------|
| plan | 58 | 11.9k | 640.2k | 72.7k | 1 | 7m 13s |
| verify | 51 | 16.1k | 3.0M | 100.7k | 1 | 8m 51s |
| implement | 252 | 16.9k | 2.3M | 83.4k | 1 | 7m 42s |
| prepare_pr | 52 | 4.3k | 172.1k | 27.8k | 1 | 1m 22s |
| compose_pr | 67 | 2.0k | 219.7k | 18.8k | 1 | 43s |
| review_pr | 198 | 38.6k | 1.2M | 70.1k | 1 | 7m 37s |
| resolve_review | 219 | 8.7k | 1.2M | 59.6k | 1 | 5m 18s |
| **Total** | 897 | 98.5k | 8.7M | 433.0k | | 38m 49s |

### CODE REVIEW

File: tests/planner/test_compiler.py [hunk 1]
```diff
@@ -237,3 +237,50 @@ def test_compile_plan_creates_issues_directory(tmp_path: Path) -> None:
...
+                    "findings": [{"message": "test", "severity": "error", "check": "test"}],
+                    "warnings": [],
+                    "schema_version": 2,
+                }
+            )
+        )
+
+        result = compile_plan(str(tmp_path), "t", "s")
+
+        assert isinstance(result, dict)
```
Comment (Trecek) [inline tests/planner/test_compiler.py]: [warning] tests: test_nonpassing_verdict_returns_dict uses only `isinstance(result, dict)` and `"plan_path" in result` as assertions — both are true on every compile_plan call regardless of verdict. The test does guard against accidental raises on fail verdicts, but cannot detect regressions in non-passing verdict behavior (e.g., missing warning log, wrong return fields). Consider asserting a more specific observable difference for the fail path.

**Review by Trecek — COMMENTED** [review]
AutoSkillit PR Review — Verdict: approved_with_comments

**Review by Trecek — COMMENTED** [review]
AutoSkillit review: warning-only findings detected. See inline comments — no blocking changes required.

File: tests/planner/test_compiler.py (hunk 1 above)
Comment (Trecek) [reply tests/planner/test_compiler.py]: Agreed — fixed in edf4fb6992d7929bde5ae4b2693543ef2f214bc2. Added structlog.testing.capture_logs() to assert the warning is emitted with verdict='fail', and replaced the two weak assertions with an exact key-set check (plan_path, plan_json_path, plan_parts) so the test now verifies the specific non-passing-verdict behavior rather than incidentally passing on any successful compile_plan call.
